### PR TITLE
[SES5] [backport] qa/tasks: py3 compatibility

### PR DESCRIPTION
**Pull request by shyukri**
Signed-off-by: Shyukri Shyukriev <shshyukriev@suse.com>
(cherry picked from commit 29f036121e24bf5b8043709c0e0d64b2a8e4bbde)

 Conflicts:
	qa/tasks/salt_manager.py

`gather_logs` was extended meanwhile...

### CODE REVIEW

**Comment by shyukri** [comment]
skip-ci

File: qa/tasks/salt_manager.py [hunk 1]
```diff
@@ -169,8 +169,26 @@ def gather_logfile(self, logfile):
...
+                raise InternalError((
+                    'Unable to harvest logs from absolute directory ->{}<- '
+                    'because no archive option was passed'
+                    ).format(logdir)
+                    )
+        else:
+            if not archive:
+                archive = logdir
+            logdir = '/var/log/{}'.format(logdir)
+        for _remote in self.ctx.cluster.remotes.keys():
```
Comment (smithfarm) [inline qa/tasks/salt_manager.py]: Adding unrelated code to the cherry-pick is not a good way to resolve the conflict. As I see it, there are two options, here. Either:

* find out which commit added this code in `ses6-downstream-commits` and cherry-pick that commit, first - i.e. make the cherry-pick apply cleanly

or

* make the `.iterkeys` -> `.keys` change manually to resolve the conflict

The first option is higher-risk (it still brings in the unrelated code, which might cause a regression). I would go with the second option.

Comment (shyukri) [reply qa/tasks/salt_manager.py]: Going with second option.

**Comment by shyukri** [comment]
Run: http://10.86.3.20:8081/ubuntu-2019-10-14_08:53:57-deepsea:tier2-wip-qa-tasks-py2-p3-ses5-backport---basic-openstack/

**Pull request merged**